plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -904,50 +907,53 @@ def main():
     except Exception as e:
         module.fail_json(msg='authorization on proxmox cluster failed with exception: %s' % e)
 
-    # If vmid not set get the Next VM id from ProxmoxAPI
-    # If vm name is set get the VM id from ProxmoxAPI
+    # If vmid is not defined then retrieve its value from the vm name,
+    # the cloned vm name or retrieve the next free VM id from ProxmoxAPI.
     if not vmid:
-        if state == 'present' and (not update and not clone) and (not delete and not revert):
+        if state == 'present' and not update and not clone and not delete and not revert:
             try:
                 vmid = get_nextvmid(module, proxmox)
             except Exception as e:
                 module.fail_json(msg="Can't get the next vmid for VM {0} automatically. Ensure your cluster state is good".format(name))
         else:
+            clone_target = name if not clone else clone
```

Comment:
In Python, you can also write:
```suggestion
            clone_target = clone or name
```